core/components/product-card/compare.tsx

```diff
@@ -21,6 +21,8 @@ export const Compare = ({
   productName: string;
 }) => {
   const checkboxId = useId();
+  const labelId = `${checkboxId}-label`;
```

Comment:
After looking at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role, I think we can remove the `htmlFor` + `checkboxId` as we should be using the pattern you are proposing here.

Reply:
Let's just rename `checkboxId` to `labelId` instead as `useId` is guaranteed to be unique.

Reply:
Good catch thanks.